From 874b5f3252cea54f9b4e8d984f16dec73fedb53f Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Sun, 21 Aug 2016 07:22:34 +0700
Subject: [PATCH 5/9] waylandsink: Ensure wayland display create queue event
 This patch ensure the complete porting from Gstreamer 1.2.3 commit
 13dcc91b4ee47b3dacc9131738902c022650425d

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 ext/wayland/wldisplay.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/wayland/wldisplay.c b/ext/wayland/wldisplay.c
index d12ddb4..0a4c8c9 100644
--- a/ext/wayland/wldisplay.c
+++ b/ext/wayland/wldisplay.c
@@ -321,6 +321,9 @@ gst_wl_display_new_existing (struct wl_display * display,
   self->own_display = take_ownership;

   self->queue = wl_display_create_queue (self->display);
+  if (!self->queue) {
+	return NULL;
+  }
   self->registry = wl_display_get_registry (self->display);
   wl_proxy_set_queue ((struct wl_proxy *) self->registry, self->queue);
   wl_registry_add_listener (self->registry, &registry_listener, self);
--
1.9.1
